src/main/java/com/epam/aidial/core/util/BufferingReadStream.java

```diff
@@ -116,7 +117,19 @@ public synchronized ReadStream<Buffer> endHandler(Handler<Void> handler) {
         return this;
     }
 
+    public synchronized void end(HttpServerResponse response) {
+        if (lastChunk != null) {
+            response.end(lastChunk);
+        } else {
+            response.end();
+        }
+    }
+
     private synchronized void handleChunk(Buffer chunk) {
+        if (lastChunk != null) {
+            // stop streaming
+            return;
+        }
```

Comment:
stop streaming  == stop calling hander

Reply:
remember position at which your stopped streaming and send in end method

Reply:
fixed